Dockerfile

```diff
@@ -12,47 +12,53 @@ ENV PATH /tmp/google-cloud-sdk/bin:$PATH
 
 WORKDIR /tmp
 
-# Buildpack-deps basic requirements
+# Install dependencies
 RUN apk add --update --no-cache \
         ca-certificates \
         curl \
+        python \
         gnupg \
         wget \
         tar \
-        unzip && \
-        apk add --update --no-cache \
+        unzip \
+        py-crcmod \
+        libc6-compat \
+        openssh-client \
         git \
-        bash && \
-        curl --location https://download.docker.com/linux/static/edge/x86_64/docker-$DOCKER_CLIENT_VERSION.tgz --output /tmp/docker-$DOCKER_CLIENT_VERSION.tgz && \
+        bash
+
+# Install Docker
+RUN curl --location https://download.docker.com/linux/static/edge/x86_64/docker-$DOCKER_CLIENT_VERSION.tgz --output /tmp/docker-$DOCKER_CLIENT_VERSION.tgz && \
```

Comment:
Not sure about separating it into distinct `RUN`s - I've seen the chaining in many different Dockerfiles (eg. `nginx`) and it produces a single Docker layer, which might come with advantages (eg. smaller image size, being up-to-date as all commands are run every time). This approach also comes with advantages (eg. quicker build as cached layers might get used, stability of installed packages as commands are not run every time).

I guess I'm OK with both approaches as you at least chain the installation with a clean-up, keeping the Docker layer size low.

Reply:
My goal here was to keep dockerfile clear, as you mentioned nginx Dockerfile, I am not really fan of that. I would keep it like this if you dont mind, but if you want, i can make it to one RUN